### PR TITLE
build fixes

### PR DESCRIPTION
**Pull request by loriab**
## Description
Accumulated fixes from Mac binary build round

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] fix some ambit import and temporarily reroute source location until @jturney around to process cmake changes
- [x] in an effort to standardize internal builds as using static libraries and detected builds as using shared, **boost detection now looks for shared**
- [x] install `psi4.so` to site-packages rather than bin
- [x] fix mollerplesset plugin
- [x] misc. cleanup
- [x] switched Intel to `-qopenmp`
- [x] added a couple include paths so plugin includes so that plugins can be compiled against installed instead of source psi4
- [x] fix imports and iwl checks in plugin templates

## Status
- [x]  Ready to go

### CODE REVIEW

File: plugins/mollerplesset2/pymodule.py [hunk 1]
```diff
@@ -28,6 +28,11 @@ def run_plugin_mp2(name, **kwargs):
     if psi4.get_global_option('REFERENCE') == 'ROHF':
         scf_wfn.semicanonicalize()
 
+    # Ensure IWL files have been written when not using DF/CD
+    scf_type = psi4.get_option('SCF', 'SCF_TYPE')
+    if ( scf_type == 'PK' or scf_type == 'DIRECT' ):
+        proc_util.check_iwl_file_from_scf_type(psi4.get_option('SCF', 'SCF_TYPE'), scf_wfn)
```
Comment (dgasmith) [inline plugins/mollerplesset2/pymodule.py]: If I remember right this function will check for this already, should be a one-liner.

Comment (loriab) [inline plugins/mollerplesset2/pymodule.py]: Thanks, I think I've fixed it in a few more places.

**Comment by dgasmith** [comment]
Good deal. Is this ready to pull in now?

**Comment by loriab** [comment]
Yes, ready to go.

**Pull request merged**